doc/api-design.md

```diff
@@ -0,0 +1,76 @@
+# API Design
+
+Before you start coding your service, you need to think hard about your API's
+design, especially for public a public service exposing its API. Below are a
+couple of practices you should follow.
+
+## Statelessness
+
+RESTful API services should be
+[stateless](https://en.wikipedia.org/wiki/Service_statelessness_principle), since
+they are conceptually modelled around *resurces* (as opposed to *systems*).
+Accordingly, your service should take actions depending on assumptions about the
+caller or the current state of the service's process (e.g. that the user is
+logged in, that they are allowed to modify a resource, etc.)
+
+## Versioning
+
+You should always version all of your API. Always. Period. Applications depend on
```

Comment:
No, I think it's fine. However you could mention the package managers. At least in JAVA dependencies to -SNAPSHOT versions create major problems. Because you never know, which version you are actually using. And maven uses the version number to decide, if a new version is dowloaded or the cached one is used. I guess npm does similar things.